### DISCUSSION TITLE
BGR linesize

### DISCUSSION THREAD

**Discussion by Yay295** [python-pillow/Pillow#6544] [Q&A] [answered]
The `linesize` of every mode except for the BGR modes is `xsize * pixelsize`. The BGR modes instead use `(xsize * pixelsize + 3) & -4`.

https://github.com/python-pillow/Pillow/blob/ad7be550aa539dc4c79da928dad071cf0150b731/src/libImaging/Storage.c#L131-L161

The file history shows that they've been like that since the fork, but does anyone know why they use a different formula?

**Comment by radarhere** [comment, marked as answer]
My guess would be that it is rounding up to a multiple of 4 for the same reason that [LA/La/PA are stored in 4 bytes](https://github.com/python-pillow/Pillow/issues/6503) - to use image32.

I'm not saying that should be the case, but it makes sense for those two things to be related.
